File: app/src/main/java/com/robotnec/budget/core/mvp/view/AddTransactionView.java
package com.robotnec.budget.core.mvp.view;

import java.util.Calendar;
import java.util.List;

import com.robotnec.budget.core.domain.Account;
import com.robotnec.budget.core.domain.Category;
import com.robotnec.budget.core.domain.MoneyAmount;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface AddTransactionView extends View {
    void initDatePickerButton(String date);

    void showAccountsChooserDialog(List<Account> accounts);

    void showCategoryChooserDialog(List<Category> categories);

    void showChosenDate(String date);

    void showDateChooserDialog(Calendar lastDate);

    void showCalculator(MoneyAmount initialAmount);

    void displayAccount(Account account);

    void displayCategory(Category category);

    void displayInitialAmount(MoneyAmount initialAmount);

    void finish();
}

File: app/src/main/java/com/robotnec/budget/core/mvp/view/AddTransactionView.kt
package com.robotnec.budget.core.mvp.view

import com.robotnec.budget.core.domain.Account
import com.robotnec.budget.core.domain.Category
import org.joda.money.Money
import java.util.Calendar

/**
 * @author zak zak@swingpulse.com>
 */
interface AddTransactionView : View {
    fun initDatePickerButton(date: String)

    fun showAccountsChooserDialog(accounts: List<Account>)

    fun showCategoryChooserDialog(categories: List<Category>)

    fun showDateChooserDialog(lastDate: Calendar)

    fun showChosenDate(date: String)

    fun showCalculator(initialAmount: Money)

    fun displayAccount(account: Account)

    fun displayCategory(category: Category)

    fun displayAmount(initialAmount: Money)

    fun finish()
}
